test/shared_utilities/utilities.jl

```diff
@@ -156,11 +157,20 @@ end
 
         # make fields non-constant in order to check the
         # impact of the dss step
-        for i in eachindex(parent(field))
-            parent(field)[i] = sin(i)
-            parent(subfield1)[i] = cos(i)
-            parent(subfield2)[i] = sin(i) * cos(i)
-        end
+        ArrayType = ClimaComms.array_type(ClimaComms.device())
```

Comment:
It might be nice to have a short comment explaining why these steps are needed (compared with the previous)